nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -41,47 +48,68 @@ abstract contract Issuer is Upgradeable {
     uint16 public currentMintingPeriod;
 
     /**
-    * @notice Constructor sets address of token contract and coefficients for mining
-    * @dev Mining formula for one stake in one period
-    (totalSupply - currentSupply) * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k2
-    if allLockedPeriods > rewardedPeriods then allLockedPeriods = rewardedPeriods
+    * @notice Constructor sets address of token contract and coefficients for minting
+    * @dev Minting formula for one sub-stake in one period for the first phase
+    maxFirstPhaseReward * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k3
+    * @dev Minting formula for one sub-stake in one period for the second phase
+    (totalSupply - currentSupply) / k2 * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k3
+    if allLockedPeriods > maxRewardedPeriods then allLockedPeriods = maxRewardedPeriods
     * @param _token Token contract
     * @param _hoursPerPeriod Size of period in hours
-    * @param _miningCoefficient Mining coefficient (k2)
-    * @param _lockedPeriodsCoefficient Locked periods coefficient (k1)
-    * @param _rewardedPeriods Max periods that will be additionally rewarded
+    * @param _secondPhaseMintingCoefficient Minting coefficient for the second phase (k2)
```

Comment:
```suggestion
    * @param _issuanceDecayCoefficient  Coefficient which modifies the rate at which the maximum issuance decays (d2), only applicable to Phase 2.  d2 = 365 * half-life / LOG2 where default half-life = 2.  See Equation 10 in Staking Protocol & Economics paper
```

Reply:
If this kind of explicit naming and description is to everyone's taste, I will do the same for others (approving this change doesn't imply approval of my forthcoming suggestions)

Reply:
I like this one :)

Reply:
✔️

Reply:
Actually it should be _d1_ because its the first and only decay coefficient

Reply:
Also I think stakers are likely to parse through Issuer.sol (rather than economics.py) to work out what affects their subsidies and then model their revenue - hence the descriptive comments here. We can repeat these in other files too if need be

Reply:
maybe name _d_? because _d1_ reads like somewhere _d2_
I put same comment in `economics.py`